tests/contributing.py

```diff
@@ -121,3 +126,65 @@ def validate_transformer_integration(transformer):
     print(tabulate(summary, headers='keys', showindex=False))
 
     return validation_error is None and error_trace is None
+
+
+def validate_transformer_performance(desired_transformer):
+    """Validate the performance of a transformer.
+
+    Run the specified Transformer on all the Dataset Generators of the indicated data type
+    and produce a report about its performance and how it compares to the other
+    Transformers of the same data type.
+
+    Args:
+        desired_transformer (string or rdt.transformers.BaseTransformer):
+            The transformer to validate.
+
+    Returns:
+        pandas.DataFrame:
+            Performance results of the transformer.
+    """
+    if isinstance(desired_transformer, str):
+        desired_transformer = get_class(desired_transformer)
+
+    print(f'Validating Performance for transformer {desired_transformer.__name__}\n')
+
+    data_type = desired_transformer.get_input_type()
+    transformers = get_transformers_by_type().get(data_type, [])
+    dataset_generators = get_dataset_generators_by_type().get(data_type, [])
+
+    total_results = pd.DataFrame(
```

Comment:
I think this loop may be too complex.
We may be able to simplify if we do not make any aggregation until the end, and during the loop we just accumulate results (see more comments further down, on the other file).

```python3
total_results = pd.DataFrame()
for transformer in transformers:
    for dataset in datasets:
        results = ...compute results for transformer and whether they are valid or not...
        total_results = total_results.append(results)

other_results = total_results[total_results.transformer != transformer.__name__]
average = other_results.groupby('Evaluation Metric')['Value'].mean()    # this may be empty if there are no others.

total_results = total_results[total_results.transformer == transformer.__name__]
final_results = total_results.groupby('Evaluation Metric').agg({
    'Value': 'mean',
    'Valid': 'any'
}).reset_index()

# From here on, `final_results` is already the output, but needs to be still formatted and cleaned.
```